From 03c08285c692862a72cc794d9fe05961ec0ceb8e Mon Sep 17 00:00:00 2001
From: Benjamin Kietzman <bengilgit@gmail.com>
Date: Fri, 14 Jun 2019 07:55:38 -0500
Subject: [PATCH] ARROW-5517: [C++] Only check header basename for 'internal'
 when collecting public headers

Author: Benjamin Kietzman <bengilgit@gmail.com>

Closes #4551 from bkietz/5517-Header-collection-CMake-logic-should-onl and squashes the following commits:

140b95b81 <Benjamin Kietzman> only check header basename for 'internal' when collecting public headers
---
 cpp/cmake_modules/BuildUtils.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cpp/cmake_modules/BuildUtils.cmake b/cpp/cmake_modules/BuildUtils.cmake
index 781cedcc2..293a7ef84 100644
--- a/cpp/cmake_modules/BuildUtils.cmake
+++ b/cpp/cmake_modules/BuildUtils.cmake
@@ -720,9 +720,11 @@ function(ARROW_INSTALL_ALL_HEADERS PATH)
 
   set(PUBLIC_HEADERS)
   foreach(HEADER ${CURRENT_DIRECTORY_HEADERS})
-    if(NOT ((HEADER MATCHES "internal")))
-      list(APPEND PUBLIC_HEADERS ${HEADER})
+    get_filename_component(HEADER_BASENAME ${HEADER} NAME)
+    if(HEADER_BASENAME MATCHES "internal")
+      continue()
     endif()
+    list(APPEND PUBLIC_HEADERS ${HEADER})
   endforeach()
   install(FILES ${PUBLIC_HEADERS} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PATH}")
 endfunction()
-- 
2.22.0

